### PR TITLE
Migration to 1ES Templates

### PR DESCRIPTION
**Pull request by payton2022**
I have migrated the pipeline for the [securities-utilities](https://github.com/microsoft/security-utilities) repo so that it now has 1ES compliance. A successful run of the template can be seen here at this [link](https://dev.azure.com/mseng/1ES/_build/results?buildId=18235025&view=results). The pipeline being run before I did this migration can be seen [here](https://dev.azure.com/mseng/1ES/_build/results?buildId=18234155&view=results) for comparison.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -5,25 +5,58 @@ variables:
...
-    arguments: '--configuration $(buildConfiguration) --no-restore'
-    
-- task: DotNetCoreCLI@2
-  displayName: 'dotnet test'
-  inputs:
-    command: test
-    projects: $(solution)
-    arguments: '--collect "Code coverage" --no-build --configuration $(buildConfiguration)'
+          steps:
+            - task: UseDotNet@2
```
Comment (sirkiza) [inline azure-pipelines.yml:30]: please use vs2019 container and remove those "UseDotNet" tasks

Comment (payton2022) [reply azure-pipelines.yml]: done

File: azure-pipelines.yml [hunk 2]
```diff
@@ -5,25 +5,56 @@ variables:
   solution: 'src\Microsoft.Security.Utilities.sln'
   buildConfiguration: 'Release'
 
-steps:
-- task: DotNetCoreCLI@2
-  displayName: 'dotnet restore'
-  inputs:
-    command: 'restore'
-    projects: $(solution)
+resources:
```
Comment (marmegh) [inline azure-pipelines.yml:8]: I've compared the  build tasks before and after this change. No lost artifacts between the two. Build tasks appear to be accomplishing everything expected for this repository. Nicely done.

**Review by marmegh — APPROVED** [review]
:shipit:

File: azure-pipelines.yml (hunk 2 above)
Comment (marmegh) [reply azure-pipelines.yml:8]: @michaelcfanning/@cfaucon, one of the two of you is required to review before this PR can be merged.

**Review by sirkiza — APPROVED** [review]

File: azure-pipelines.yml [hunk 3]
```diff
@@ -5,25 +5,56 @@ variables:
...
+        jobs:
+          - job: build
+            templateContext:
+              inputs:
+              - input: checkout
+                repository: self
+                submodules: true
+              outputs:
+              - output: pipelineArtifact
+                targetPath: $(System.DefaultWorkingDirectory)
```
[the remark this answers is not in the document]
Comment (payton2022) [reply azure-pipelines.yml:38]: set an artifact name here

**Review by eddynaka — APPROVED** [review]

**Review by cfaucon — APPROVED** [review]

**Pull request merged**